### PR TITLE
83505: Add hexapdf for PDF decryption in Decision Review evidence flow

### PR DESCRIPTION
**Pull request by dfong-adh**
## Summary
This PR introduces the `Hexapdf` gem as it can be used for PDFs with newer encryption methods (i.e. AES-256) unlike `pdftk`. A flipper flag named `decision_review_use_hexapdf_for_encrypted_attachments` can toggle between the new library and the existing tool (`pdftk`) for Decision Review evidence attachment decryption.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/83503

## Testing done
- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
 Decrypted PDFs used `pdftk` for decryption
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
  Manual testing was performed to validate the decryption was working
- *If this work is behind a flipper:*
  - [x] *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*
  This will be verified and monitored in staging, and then enabled and monitored in production. In the case any issues arise, the flipper flag can be used to disable the new code flow.

## What areas of the site does it impact?
This impacts the Decision Review Evidence Attachment uploads to S3.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1336,6 +1337,7 @@ spec/lib/common/client/concerns/mhv_fhir_session_client_spec.rb @department-of-v
 spec/lib/common/client/concerns/mhv_jwt_session_client_spec.rb @department-of-veterans-affairs/vfs-mhv-medical-records @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
 spec/lib/common/client/concerns/mhv_locked_session_client_spec.rb @department-of-veterans-affairs/vfs-mhv-medical-records @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
 spec/lib/common/client/middleware @department-of-veterans-affairs/backend-review-group
+spec/lib/common/pdf_helpers_spec.rb department-of-veterans-affairs/benefits-decision-reviews-be @department-of-veterans-affairs/backend-review-group
```
Comment (anniebtran) [inline .github/CODEOWNERS:1340]:
```suggestion
spec/lib/common/pdf_helpers_spec.rb @department-of-veterans-affairs/benefits-decision-reviews-be @department-of-veterans-affairs/backend-review-group
```

File: lib/common/pdf_helpers.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+module Common
+  module PdfHelpers
+    extend SentryLogging
+
+    def self.unlock_pdf(input_file, password, output_file)
+      doc = HexaPDF::Document.open(input_file, decryption_opts: { password: })
+      doc.encrypt(name: nil)
+      doc.write(output_file)
+    rescue HexaPDF::EncryptionError => e
+      log_message_to_sentry(e.message, 'warn')
```
Comment (anniebtran) [inline lib/common/pdf_helpers.rb:15]: I wonder if we should also add a `Rails.logger.<info || error || etc>` so we can monitor these issues from DataDog too? 🤔

File: spec/controllers/v0/decision_review_evidences_controller_spec.rb [hunk 3]
```diff
@@ -161,6 +161,70 @@
           expect(JSON.parse(response.body)).to eq(expected_response_body)
         end
       end
+
+      context 'and the use hexapdf feature switch is disabled' do
+        before { Flipper.disable :decision_review_use_hexapdf_for_encrypted_attachments }
+
+        it 'creates a FormAttachment, logs formatted error, and increments statsd' do
```
Comment (anniebtran) [inline spec/controllers/v0/decision_review_evidences_controller_spec.rb:168]:
```suggestion
        it 'creates a FormAttachment, logs formatted success message, and increments statsd' do
```
looks like this covers the success case based on `expect(Rails.logger).to receive(:info).with(encrypted_log_params_success)`

File: spec/controllers/v0/decision_review_evidences_controller_spec.rb [hunk 4]
```diff
@@ -161,6 +161,70 @@
...
+
+          expect(response).to have_http_status(:ok)
+          expect(JSON.parse(response.body)).to eq(expected_response_body)
+        end
+      end
+
+      context 'and the use hexapdf feature switch is enabled' do
+        before { Flipper.enable :decision_review_use_hexapdf_for_encrypted_attachments }
+
+        it 'creates a FormAttachment, logs formatted error, and increments statsd' do
```
Comment (anniebtran) [inline spec/controllers/v0/decision_review_evidences_controller_spec.rb:200]:
```suggestion
        it 'creates a FormAttachment, logs formatted success message, and increments statsd' do
```

File: .github/CODEOWNERS (hunk 1 above)
Comment (dfong-adh) [reply .github/CODEOWNERS]: Thanks for catching that 👍

**Review by anniebtran — APPROVED** [review]
Great work! 🎉

File: Gemfile [hunk 5]
```diff
@@ -86,6 +86,7 @@ gem 'googleauth'
 gem 'google-protobuf' # For Datadog Profiling
 gem 'govdelivery-tms', git: 'https://github.com/department-of-veterans-affairs/govdelivery-tms-ruby.git', tag: 'v4.0.0', require: 'govdelivery/tms/mail/delivery_method'
 gem 'gyoku'
+gem 'hexapdf'
```
Comment (ericboehs) [inline Gemfile:89]: I believe this also needs the hexapdf package added to the Dockerfile?

**Review by ericboehs — APPROVED** [review]

File: Gemfile (hunk 5 above)
Comment (dfong-adh) [reply Gemfile:89]: Thanks for reviewing - is there any documentation for how to update the Dockerfile? This is my first time adding a new gem to vets-api so any guidance here would be very helpful.

**Pull request merged**